test/alphabet/nucleotide_test.cpp

```diff
@@ -0,0 +1,216 @@
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <gtest/gtest.h>
+#include <sstream>
+
+#include <seqan3/alphabet/nucleotide.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+template <typename T>
+class nucleotide : public ::testing::Test
+{};
+
+// add all alphabets from the nucleotide sub module here
+using nucleotide_types = ::testing::Types<dna4, dna5, rna4, rna5, nucl16>;
+
+TYPED_TEST_CASE(nucleotide, nucleotide_types);
+
+TYPED_TEST(nucleotide, static_members)
+{
+    TypeParam::A;
+    TypeParam::C;
+    TypeParam::G;
+    TypeParam::T;
+    TypeParam::U;
+    TypeParam::UNKNOWN;
+}
+
+TYPED_TEST(nucleotide, assign_char)
+{
+    EXPECT_EQ((assign_char(TypeParam{}, 'A')), TypeParam::A);
+    EXPECT_EQ((assign_char(TypeParam{}, 'C')), TypeParam::C);
+    EXPECT_EQ((assign_char(TypeParam{}, 'G')), TypeParam::G);
+    EXPECT_EQ((assign_char(TypeParam{}, 'T')), TypeParam::T);
+    EXPECT_EQ((assign_char(TypeParam{}, 'U')), TypeParam::U);
+    EXPECT_EQ((assign_char(TypeParam{}, '!')), TypeParam::UNKNOWN);
+}
+
+TYPED_TEST(nucleotide, to_char)
+{
+    EXPECT_EQ(to_char(TypeParam::A), 'A');
+    EXPECT_EQ(to_char(TypeParam::C), 'C');
+    EXPECT_EQ(to_char(TypeParam::G), 'G');
+
+    if constexpr (std::is_same_v<TypeParam, rna4> || std::is_same_v<TypeParam, rna5>)
+    {
+        EXPECT_EQ(to_char(TypeParam::U), 'U');
+        EXPECT_EQ(to_char(TypeParam::T), 'U');
+    }
+    else if constexpr (std::is_same_v<TypeParam, dna4> || std::is_same_v<TypeParam, dna5>)
+    {
+        EXPECT_EQ(to_char(TypeParam::U), 'T');
+        EXPECT_EQ(to_char(TypeParam::T), 'T');
+    } else
+    {
+        EXPECT_EQ(to_char(TypeParam::U), 'U');
+        EXPECT_EQ(to_char(TypeParam::T), 'T');
+    }
```

Comment:
This test is for the entire nucleotide submodule, but not for compositions. None of the tests work for compositions, because compositions don't provide the enum-like static members.
But maybe I should make this clearer somewhere.